chainer/functions/math/trigonometric.py

```diff
@@ -67,3 +67,39 @@ def backward_gpu(self, x, gy):
 def cos(x):
     """Elementwise cos function."""
     return Cos()(x)
+
+
+class Tan(function.Function):
+
+    @property
+    def label(self):
+        return 'tan'
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        type_check.expect(in_types[0].dtype.kind == 'f')
+
+    def forward(self, x):
+        xp = cuda.get_array_module(*x)
+        return utils.force_array(xp.tan(x[0])),
+
+    def backward_cpu(self, x, gy):
+        gx = utils.force_array(numpy.cos(x[0]))
+        numpy.square(gx, out=gx)
+        numpy.reciprocal(gx, out=gx)
+        gx *= gy[0]
+        return gx,
+
+    def backward_gpu(self, x, gy):
+        gx = cuda.elementwise(
+            'T x, T gy',
+            'T gx',
+            'gx = 1.0 / pow(cos(x), (T)2.0) * gy',
```

Comment:
Here I hope `pow()` is optimized away to a multiplication, I'm not sure actually yet.